rbac/management/cache.py

```diff
@@ -34,41 +34,122 @@ def connection(self):
                 raise
         return self._connection
 
+    @contextlib.contextmanager
+    def delete_handler(self, err_msg):
+        """Handle delete events."""
+        try:
+            yield
+        except exceptions.RedisError:
+            logger.exception(err_msg)
+
+    def get_from_redis(self, key):
+        """Get object from redis based on key."""
+        raise Exception("Please override the get_from_redis method.")
+
+    def get_cached(self, key, error_message):
+        """Get cached object from redis, throw error if there is any."""
+        try:
+            return self.get_from_redis(key)
+        except exceptions.RedisError:
+            logger.exception(error_message)
+        return None
+
+    def delete_cached(self, key, obj_name):
+        """Delete cache from redis."""
+        err_msg = f"Error deleting {obj_name} for {key}"
+        with self.delete_handler(err_msg):
+            logger.info(f"Deleting {obj_name} cache for {key}")
+            self.connection.delete(self.key_for(key))
+
+    def set_cache(self, pipe, key, item):
+        """Set cache to redis."""
+        raise Exception("Please override the set_cache method.")
```

Comment:
Love the pattern, but a small nit is that we may want to use `NotImplementedError` in these cases?

Reply:
Thanks! This is what I was looking for